### PR TITLE
Add k6 benchmark to suite

### PR DESCRIPTION
**Pull request by jeffa5**
We had it lying around anyway, might as well wire it up properly!

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,8 +5,16 @@ cmake_minimum_required(VERSION 3.16)
...
+option(CCF_UNSAFE "build with unsafe options" OFF)
+if(CCF_UNSAFE)
+  message(WARNING "Building with unsafe options")
+  if(NOT TARGET ccf_unsafe)
+    find_package(ccf_unsafe REQUIRED)
+  endif()
+else()
+  if(NOT TARGET ccf)
+    find_package(ccf REQUIRED)
+  endif()
```
Comment (jumaffre) [inline CMakeLists.txt:17]:
```suggestion
find_package(ccf_unsafe REQUIRED)
else()
find_package(ccf REQUIRED)
```

Should also work I believe. The extras `if(NOT TARGET...)` are only there as a leftover from the logging app in the CCF repo that should be buildable both standalone and in the CCF build.

File: Makefile [hunk 2]
```diff
@@ -53,11 +61,23 @@ debug-dockerignore:
 
 .PHONY: run-virtual
 run-virtual: build-virtual
-	$(CCF_PREFIX)/bin/sandbox.sh -p $(BUILD)/liblskv.virtual.so --http2
+	VENV_DIR=.venv $(CCF_PREFIX)/bin/sandbox.sh -p $(BUILD)/liblskv.virtual.so --http2
+
+.PHONY: run-virtual-unsafe
```
Comment (jumaffre) [inline Makefile:66]: For 3.0.0, we will only publish an unsafe package for the sgx target.

Comment (jeffa5) [reply Makefile:66]: Will there be a way to get the verbose CCF logs in virtual mode?

File: patches/k6-micro.diff [hunk 3]
```diff
@@ -0,0 +1,13 @@
+diff --git a/output/csv/output.go b/output/csv/output.go
+index 582ad7e5..c4e94ba2 100644
+--- a/output/csv/output.go
++++ b/output/csv/output.go
+@@ -198,7 +198,7 @@ func SampleToRow(sample *metrics.Sample, resTags []string, ignoredTags []string,
+ 	case TimeFormatRFC3339:
+ 		row[1] = sample.Time.Format(time.RFC3339)
+ 	case TimeFormatUnix:
+-		row[1] = strconv.FormatInt(sample.Time.Unix(), 10)
++		row[1] = strconv.FormatInt(sample.Time.UnixMicro(), 10)
```
Comment (jumaffre) [inline patches/k6-micro.diff:10]: Is this to get more precision out of the k6 output? I would have thought they'd have a high precision format by default?

File: Makefile (hunk 2 above)
Comment (jumaffre) [reply Makefile:66]: That's a good point. We could simply add a runtime parameter for the enclave logging in virtual mode (it is a build-time parameter in SGX as logging level should change enclave measurement for security reasons).

File: patches/k6-micro.diff (hunk 3 above)
Comment (jeffa5) [reply patches/k6-micro.diff:10]: This is, for the CSV raw output they only output second-resolution timestamps without this change.

I don't think the `timePrecision` part of this comment has been added yet: https://github.com/grafana/k6/pull/2274#issuecomment-1085781856

File: Makefile (hunk 2 above)
Comment (jumaffre) [reply Makefile:66]: https://github.com/microsoft/CCF/issues/4441

File: .gitmodules [hunk 4]
```diff
@@ -4,3 +4,6 @@
 [submodule "3rdparty/go-ycsb"]
 	path = 3rdparty/go-ycsb
 	url = https://github.com/pingcap/go-ycsb.git
+[submodule "3rdparty/k6"]
```
Comment (jumaffre) [inline .gitmodules:7]: Could you please add this to the `cfmanifest.json` file as well?

File: benchmark/all.py [hunk 5]
```diff
@@ -66,11 +68,24 @@ def all_perf_configurations(args: argparse.Namespace) -> List[perf_system.PerfCo
     Set args for all perf configurations.
     """
     args.http1 = True
+    args.http2 = True
```
Comment (jumaffre) [inline benchmark/all.py:71]: What does both set to true mean in this case? I'm assuming the benchmark is run twice for both HTTP/1 and HTTP/2 by default?

**Review by jumaffre — APPROVED** [review]

**Pull request merged**